modules/lang-painless/src/main/java/org/elasticsearch/painless/ir/IRNode.java

```diff
@@ -19,15 +19,85 @@
 
 package org.elasticsearch.painless.ir;
 
-import org.elasticsearch.painless.ClassWriter;
 import org.elasticsearch.painless.Location;
-import org.elasticsearch.painless.MethodWriter;
 import org.elasticsearch.painless.phase.IRTreeVisitor;
 import org.elasticsearch.painless.symbol.WriteScope;
 
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
 public abstract class IRNode {
 
-    /* ---- begin node data ---- */
+    /* ---- begin decorations ---- */
+
+    public interface IRDecoration {
+
+    }
+
+    private final Map<Class<? extends IRDecoration>, IRDecoration> decorations = new HashMap<>();
+
+    @SuppressWarnings("unchecked")
```

Comment:
This doesn't seem relevant to this PR.

Reply:
Yup, this is for the next PR. (Oops!) Removed.